src/js/stores/MesosStateStore.js

```diff
@@ -70,7 +73,7 @@ class MesosStateStore extends GetSetBaseStore {
     const getMasterRequest = request(
       { type: "GET_MASTER" },
       "/mesos/api/v1?get_master"
-    ).retry(3);
+    ).retryWhen(linearBackoff(MAX_RETRIES, RETRY_DELAY));
```

Comment:
So I am still asking myself, couldn't we have this rxjsUtils as part of the mesos client. I  think it would even make sense to move this getMasterRequest as a stream that we could also import from mesos client. Maybe it could also expose rxjs utils, and export the linearBackoff so we can use somewhere else too?

Reply:
Or maybe even we want to put on a new rxjsUtil packages, in this case, I believe we would merge here first and leave a todo to extract if/when it spreads.

Reply:
The intention behind the `mesos-client` is to provide a pair of primitives one could build upon. Those are just `request` and `stream` as this is everything Mesos provides according to the [operator api documentation](mesos.apache.org/documentation/latest/operator-http-api/). It's a bare bones solution and because of that it is not opinionated. 

Down the road I'd like seeing the [suggested backoff strategy](http://mesos.apache.org/documentation/latest/operator-http-api/#heartbeat) inside the `mesos-client` as an option. But I don't really thing `rxjsUtils` with all the strategies belongs there.

Feel free to challenge me 👍